src/main/java/oracle/r2dbc/impl/UserDefinedType.java

```diff
@@ -0,0 +1,23 @@
+package oracle.r2dbc.impl;
```

Comment:
I think the need for exporting the impl package came from here.
On the Issue thread, I had suggested a factory method on OracleR2dbcTypes. We can make UserDefinedType an inner class of OracleR2dbcTypes, also private and static. We can then have the factory method return an instance of this class. This would eliminate the need to export the impl package.
A factory method will also allow us to return an instance of the Type interface, rather than a specific concrete class. By returning an interface, we will have more freedom to change the code later without breaking existing code that calls the method.